src/ffi/ffiexports.cpp

```diff
@@ -43,15 +47,19 @@ rust::string publishEthTransaction(rust::Vec<uint8_t> rawTransaction) {
     // TODO Replace execTestTx with non-throwing function
     try {
         execTestTx(CTransaction(rawTx), targetHeight, optAuthTx);
-        send(MakeTransactionRef(std::move(rawTx)), optAuthTx)->GetHash().ToString();
+        uint256 txHash = send(MakeTransactionRef(std::move(rawTx)), optAuthTx)->GetHash();
+        UInt256 result;
+        BRKeccak256(&result, std::data(message.evmTx), message.evmTx.size());
+        auto evmTxHash = uint256S(u256hex(result));
+        pcustomcsview->SetTxHash(CEvmDvmMapType::DvmEvm, txHash, evmTxHash);
```

Comment:
Should not be added to via global. Should be done via a cache that is part of the undo system.